chain/mempool.go

```diff
@@ -12,6 +12,69 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+// cachedInputs caches the inputs of the transactions in the mempool. This is
+// used to provide fast lookup between txids and inputs.
+type cachedInputs struct {
+	// inputs provides a fast lookup from input -> txid.
+	inputs map[wire.OutPoint]chainhash.Hash
+
+	// txids provides a fast lookup from txid -> inputs.
+	txids map[chainhash.Hash]map[wire.OutPoint]struct{}
```

Comment:
In theory the second level here can just be an index? Since the txid is the first part of the outpoint.

Reply:
this map stores `spendingTxid: inputs being spent`, so the outpoint will be different from the txid.